### PR TITLE
Add test package lpdraw

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
This PR adds a test package `lpdraw`.

```bash
python integration_tests/test_pkg_lpdraw.py > cpy_tmp.pgm
```
cpy_tmp

```bash
lpython integration_tests/test_pkg_lpdraw.py --backend wasm > wasm_tmp.pgm
```
wasm_tmp

(Note: Images converted to `jpeg` before uploading here. It seems GitHub does not support uploading `.pgm` image files)

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
It seems that arrays declared as globals are not yet supported.
```bash
(lp) lpython$ cat examples/expr2.py 
from lpython import i32, Const
from numpy import empty, int32

Width: Const[i32] = 200
Height: Const[i32] = 100
Screen: i32[Height, Width] = empty((Height, Width), dtype=int32)

def main0():
    Screen[1, 1] = 5
    print(Screen[1, 1])

main0()
(lp) lpython$ lpython examples/expr2.py
Assertion failed: (Ty && "Invalid GetElementPtrInst indices for type!"), function checkGEPType, file Instructions.h, line 897.
Traceback (most recent call last):
  File "/Users/ubaid/conda_root/envs/lp/include/llvm/IR/Instructions.h", line 1074
    Type *PtrTy = PointerType::get(checkGEPType(getIndexedType(ElTy, IdxList)),
  File "/Users/ubaid/conda_root/envs/lp/include/llvm/IR/Instructions.h", line 897
    assert(Ty && "Invalid GetElementPtrInst indices for type!");
  Binary file "/usr/lib/system/libsystem_c.dylib", local address: 0x18030561f
  Binary file "/usr/lib/system/libsystem_c.dylib", local address: 0x1803062c7
  Binary file "/usr/lib/system/libsystem_pthread.dylib", local address: 0x1803ccceb
  Binary file "/usr/lib/system/libsystem_platform.dylib", local address: 0x1803fb2a3
Abort: Signal SIGABRT (abort) received
(lp) lpython$ 
```

**Comment by Shaikh-Ubaid** [comment]
For Commit [PKG: Add lpdraw test package](https://github.com/lcompilers/lpython/pull/1639/commits/9760c01d5d7c0a94e145511129d95f75e0b57475), output with CPython:
```
(lp) lpython$ cat integration_tests/test_pkg_lpydraw.py 
from lpydraw import Line, Circle, Clear, Display, DisplayTerminal

def main():
    Clear()
    Line(2, 4, 99, 11)
    Line(0, 39, 49, 0)
    Circle(52, 20, 6.0)
    Display()
    # DisplayTerminal()

main()
(lp) lpython$ python integration_tests/test_pkg_lpydraw.py > tmp.pgm
(lp) lpython$ 
```
tmp

For commit [temporary changes to work with wasm backend](https://github.com/lcompilers/lpython/pull/1639/commits/a55f7526257ec18ae46b489f63dd89f20fca3d9e), output with `wasm` backend:

```bash
(lp) lpython$ cat integration_tests/test_pkg_lpydraw.py 
from lpydraw import Line, Circle, Clear, Display, DisplayTerminal
from lpython import i32
from numpy import empty, int32

def main():
    Width: i32 = 100 # x-axis limits [0, 99]
    Height: i32 = 40 # y-axis limits [0, 39]
    Screen: i32[40, 100] = empty((40, 100), dtype=int32)

    Clear(Height, Width, Screen)
    Line(Height, Width, Screen, 2, 4, 99, 11)
    Line(Height, Width, Screen, 0, 39, 49, 0)
    Circle(Height, Width, Screen, 52, 20, 6.0)
    Display(Height, Width, Screen)
    # DisplayTerminal(Height, Width, Screen)

main()
(lp) lpython$ lpython integration_tests/test_pkg_lpydraw.py --backend wasm > wasm_tmp.pgm

(node:98906) ExperimentalWarning: WASI is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(lp) lpython$ 
```

wasm_tmp

(Note: Images converted to `jpeg` before uploading here. It seems `GitHub` does not support uploading `.pgm` files)

**Comment by Shaikh-Ubaid** [comment]
For `llvm`, it currently outputs:
```bash
...
declare void @_lpython_set_argv(i32, i8**)
code generation error: asr_to_llvm: module failed verification. Error:
Branch condition is not 'i1' type!
  br i32 %34, label %then, label %else
  %34 = load i32, i32* %33, align 4



Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
```

**Comment by Shaikh-Ubaid** [comment]
This is ready. Please review and share feedback.

**Review by czgdp1807 — APPROVED** [review]
This is good.

**Comment by czgdp1807** [comment]
Can you show the output with LLVM backend as well?

File: integration_tests/lpdraw/draw.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    Screen[y, x] = 255
+
+def Clear(H: i32, W: i32, Screen: i32[H, W]):
+    i: i32
+    j: i32
+    for i in range(H):
+        for j in range(W):
+            Screen[i, j] = 0
+
+def DisplayTerminal(H: i32, W: i32, Screen: i32[H, W]):
```
Comment (czgdp1807) [inline integration_tests/lpdraw/draw.py:17]: Add a function to check if the pixels in `Screen` array as expected (using asserts). Like you can add a moderately strong test by summing up all the pixels in `Screen` array and then checking that the sum value is as expected.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Review by certik — APPROVED** [review]
Beautiful, that's exactly what we need!

**Comment by Shaikh-Ubaid** [comment]
>Can you show the output with LLVM backend as well?

Sure, Please find below:

**Image:** (`DisplayTerminal()` commented out for generating image)

`lpython integration_tests/test_pkg_lpdraw.py --backend llvm > llvm_tmp.pgm`

llvm_tmp
(Converted to `jpeg` for uploading here)

**Terminal:** (`Display()` commented out)

`lpython integration_tests/test_pkg_lpdraw.py --backend llvm > stdout`

```bash
+----------------------------------------------------------------------------------------------------+
|                                                 .                                                  |
|                                                .                                                   |
|                                              ..                                                    |
|                                             .                                                      |
|  .......                                   .                                                       |
|         ..............                    .                                                        |
|                       ..............    ..                                                         |
|                                     ..............                                                 |
|                                       .           ..............                                   |
|                                      .                          ..............                     |
|                                    ..                                         ..............       |
|                                   .                                                         .......|
|                                  .                                                                 |
|                                 .                                                                  |
|                               ..                   .                                               |
|                              .                   .. ..                                             |
|                             .                   .     .                                            |
|                            .                   .       .                                           |
|                          ..                   .         .                                          |
|                         .                     .         .                                          |
|                        .                     .           .                                         |
|                      ..                       .         .                                          |
|                     .                         .         .                                          |
|                    .                           .       .                                           |
|                   .                             .     .                                            |
|                 ..                               .. ..                                             |
|                .                                   .                                               |
|               .                                                                                    |
|              .                                                                                     |
|            ..                                                                                      |
|           .                                                                                        |
|          .                                                                                         |
|         .                                                                                          |
|       ..                                                                                           |
|      .                                                                                             |
|     .                                                                                              |
|    .                                                                                               |
|  ..                                                                                                |
| .                                                                                                  |
|.                                                                                                   |
+----------------------------------------------------------------------------------------------------+
```

File: integration_tests/lpdraw/draw.py (hunk 1 above)
Comment (Shaikh-Ubaid) [reply integration_tests/lpdraw/draw.py:17]: Done. Thank you so much for the guidance!

Comment (Shaikh-Ubaid) [reply integration_tests/lpdraw/draw.py]: Sorry, not yet done. Facing some issue with pushing the changes.

**Comment by Shaikh-Ubaid** [comment]
Closing in favour of https://github.com/lcompilers/lpython/pull/1647.

**Pull request closed without merging**